### ISSUE TITLE
Profile.update() broken

### ISSUE THREAD

**Issue by pjones-d** [lxc/pylxd#110]
_Note: I haven't really used github before so let me know if I'm doing something wrong._

Profile.update() is broken due to the 'config' slot not being filled out by the profile.all() method, which causes getattr() in the 'Marshallable' mixin to break. Using git clone of master from [05/27/16] on Ubuntu 16.04 x64.

Sample code:
```
from pylxd.client import Client

API = Client()

for x in API.profiles.all():
    print(">> Name: " + x.name)
    x.update()
```
Results:
```
(env) pjones@ThinkPad:~/Development/Node-Z/src$ python nzhd.py
>> Name: default
>> NAME: _client
>> NAME: config
Traceback (most recent call last):
  File "nzhd.py", line 11, in <module>
    x.update()
  File "/home/pjones/Development/Node-Z/env/lib/python3.5/site-packages/pylxd/profile.py", line 68, in update
    marshalled = self.marshall()
  File "/home/pjones/Development/Node-Z/env/lib/python3.5/site-packages/pylxd/mixin.py", line 37, in marshall
    marshalled[name] = getattr(self, name)
AttributeError: config
```
Strangely enough I don't seem to have a profile called '_client' or 'config'.
```
(env) pjones@ThinkPad:~/Development/Node-Z/src$ lxc profile list
default
docker
Jail
nz-01
```
And when I modify the code to remove the 'x.update()' line, the profile list is accurate:

```
from pylxd.client import Client

API = Client()

for x in API.profiles.all():
    print(">> Name: " + x.name)
```

```
(env) pjones@ThinkPad:~/Development/Node-Z/src$ python nzhd.py
>> Name: default
>> Name: docker
>> Name: Jail
>> Name: nz-01
```

**Comment by rockstar** [comment]
Ah, I see what's happening here. `all` doesn't fetch a full profile (or any other object fetched with `all`), and so you shouldn't really be able to update it without getting the rest of the object. I'll take a whack at this in the morning.

**Comment by rockstar** [comment]
Fixed in #112
